pyomo/contrib/example/tests/test_example.py

```diff
@@ -0,0 +1,25 @@
+#
+# Only run the tests in this package if the TEST_PYOMO_CONTRIB environment
+# variable has been set to '1'.  Otherwise, skip the tests.
+#
+import os
+import sys
+if os.environ.get('TEST_PYOMO_CONTRIB', 0):
```

Comment:
It looks like you're putting the responsibility of checking this environment variable on the code contributor. Wouldn't it make more sense to include this check in the main pyomo testing script to determine if the testing folders in pyomo.contrib are even descended into? I would be worried about a malicious/novice contributor not including this logic in their own code and a core developer forgetting about this requirement before merging a PR.

Reply:
@blnicho Yes, there is a risk that TEST_PYOMO_CONTRIB is not appropriately checked.  But contributions in pyomo.contrib will be reviewed by Pyomo developers, which mitigates that risk.  This process is compatible with the Python ``nose`` tool;  I couldn't think of a strategy that would work with ``nose`` in the manner you suggest.

@jsiirola I think we should not run the contrib tests by default.  That would impede normal Pyomo development.  But, rather, we want to be able to enable these tests as part of our Travis/appveyor tests to check whether the packages are being maintained.